migration/usage-metric-migration/src/main/java/io/camunda/migration/usagemetric/TUMetricMigrator.java

```diff
@@ -0,0 +1,288 @@
+  private final TasklistMigrationRepositoryIndex tasklistMigrationRepositoryIndex;
+  private final MigrationConfiguration configuration;
+  private final Instant timeout;
+  private final TasklistImportPositionIndex tasklistImportPositionIndex;
+
+  public TUMetricMigrator(
+      final ConnectConfiguration connectConfiguration,
+      final MeterRegistry meterRegistry,
+      final MigrationProperties properties) {
+    metricRegistry = new MetricRegistry(meterRegistry);
+    final boolean isElasticsearch =
+        connectConfiguration.getTypeEnum() == DatabaseType.ELASTICSEARCH;
+    configuration = properties.getMigrationConfiguration(getClass());
+    client =
+        isElasticsearch
+            ? new ElasticsearchUsageMetricMigrationClient(
+                connectConfiguration, configuration.getRetry())
+            : new OpensearchUsageMetricMigrationClient(
+                connectConfiguration, configuration.getRetry());
+    indexDescriptors = new IndexDescriptors(connectConfiguration.getIndexPrefix(), isElasticsearch);
+    tasklistMigrationRepositoryIndex =
+        new TasklistMigrationRepositoryIndex(
+            connectConfiguration.getIndexPrefix(), isElasticsearch);
+    tasklistImportPositionIndex =
+        new TasklistImportPositionIndex(connectConfiguration.getIndexPrefix(), isElasticsearch);
+    timeout = Instant.now().plus(configuration.getTimeout());
+  }
+
+  @Override
+  public Void call() throws Exception {
+    LOG.info("Starting TU metrics migrator");
+    try {
+      final var reindexTask =
+          client.findOne(
+              tasklistMigrationRepositoryIndex.getFullQualifiedName(),
+              tasklistMigratorStepQuery(),
+              ProcessorStep.class);
+      var taskId = reindexTask != null ? reindexTask.getContent() : null;
+      final var completed = reindexTask != null && reindexTask.isApplied();
+      if (reindexTask == null) {
+        LOG.info("No TU migration job found, starting new migration");
+        busyWaitForImporter();
+        taskId = reindex();
+      } else {
+        if (completed) {
+          LOG.info("TU migration job already completed, nothing to do");
+          return null;
+        } else {
+          final var status = client.getTask(taskId);
+          LOG.info("TU migration step exists {}, status {}", reindexTask, status);
+          if (status.found() && status.completed()) {
+            LOG.info("TU migration job {} already completed, nothing to do", taskId);
+            client.persistMigratorStep(
+                tasklistMigrationRepositoryIndex.getFullQualifiedName(),
+                TASKLIST_MIGRATOR_STEP_ID,
+                status.taskId(),
+                TASKLIST_STEP_DESCRIPTION,
+                true);
+            return null;
+          } else {
+            LOG.info(
+                "TU migration job {}, is not present in secondary storage, restarting migration",
+                taskId);
+            taskId = reindex();
```

Comment:
👍 Nice, I think I need to refactor the other one similarly. That one does not recheck the taskId.

To be honest the call itself could be pretty much the same for both operate and tasklist migrators. The only difference is the `reindex` method itself.